src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -2106,7 +2118,128 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VSMMetadata,
     hConf.writeTextFile(dirname + "/metadata.json.gz")(Serialization.writePretty(json, _))
   }
 
+  def coalesce(k: Int, shuffle: Boolean = true): VariantSampleMatrix[RPK, RK, T] =
+    copy(rdd = rdd.coalesce(k, shuffle = shuffle)(null).toOrderedRDD)
+
+  def persist(storageLevel: String = "MEMORY_AND_DISK"): VariantSampleMatrix[RPK, RK, T] = {
+    val level = try {
+      StorageLevel.fromString(storageLevel)
+    } catch {
+      case e: IllegalArgumentException =>
+        fatal(s"unknown StorageLevel `$storageLevel'")
+    }
+
+    copy(rdd = rdd.persist(level))
```

Comment:
We've lost the genotypeStream optimization here. That was pretty important - if you do a `filter_samples` to 1% of your samples and then persist with this method, you end up storing everything on disk given the lazy filter implementation